packages/signers/src/__typetests__/sign-transaction-typetest.ts

```diff
@@ -80,3 +87,17 @@ type CompilableTransactionMessageWithSigners = CompilableTransactionMessage & IT
         ITransactionMessageWithSingleSendingSigner;
     signAndSendTransactionMessageWithSigners(transactionMessage) satisfies Promise<SignatureBytes>;
 }
+
+{
+    // [signAndSendTransactionMessageWithSigners]: it only works if the message contains a single sending signer.
+    const signer = null as unknown as TransactionSigner;
+    const latestBlockhash = null as unknown as Parameters<typeof setTransactionMessageLifetimeUsingBlockhash>[0];
+    const message = setTransactionMessageLifetimeUsingBlockhash(
+        latestBlockhash,
+        setTransactionMessageFeePayerSigner(signer, createTransactionMessage({ version: 0 })),
+    );
+    // @ts-expect-error The message could have multiple sending signers.
+    signAndSendTransactionMessageWithSigners(message);
+    assertIsTransactionMessageWithSingleSendingSigner(message);
+    signAndSendTransactionMessageWithSigners(message);
```

Comment:
This might be an awful idea, but should we call it `signAndSendTransactionMessageWithSingleSendingSigner`?

I'm not sure if that's more clear because it specifies what it needs, or less clear because there can be multiple signers and because it's obnoxiously long.

Reply:
I don't know that this would help much because if the user doesn't use the assert function, it will still end up getting the weird and unhelpful TypeScript error described in #2818 which will confuse them regardless of the function name.

Ideally, we should figure out why TS is not throwing the appropriate error but I'm really struggling with that.